pkg/reconciler/ingress/resources/virtual_service_test.go

```diff
@@ -168,8 +168,6 @@ func TestMakeMeshVirtualServiceSpec_CorrectRoutes(t *testing.T) {
 			Rules: []v1alpha1.IngressRule{{
 				Hosts: []string{
 					"test-route.test-ns.svc.cluster.local",
-					"test-route.test-ns.svc",
```

Comment:
Does this test need an update?

Reply:
This unit tests calls `makeVirtualServiceSpec` and  create `v1alpha3.HTTPMatchRequest`  with intersection hosts of expanded hosts and hosts in ingress rule.
https://github.com/knative/serving/blob/399abc270f1c664b5bac293c9722b62da2a54f91/pkg/reconciler/ingress/resources/virtual_service.go#L174
https://github.com/knative/serving/blob/399abc270f1c664b5bac293c9722b62da2a54f91/pkg/reconciler/ingress/resources/virtual_service.go#L222
As this PR expanded hosts for `-mesh` VirtualServices, for this case, there is two additional
`v1alpha3.HTTPMatchRequest` generated as following.
```
Match: []v1alpha3.HTTPMatchRequest{{
			URI:       &istiov1alpha1.StringMatch{Regex: "^/pets/(.*?)?"},
			Authority: &istiov1alpha1.StringMatch{Regex: `^test-route\.test-ns(?::\d{1,5})?$`},
			Gateways:  []string{},
		}, {
			URI:       &istiov1alpha1.StringMatch{Regex: "^/pets/(.*?)?"},
			Authority: &istiov1alpha1.StringMatch{Regex: `^test-route\.test-ns\.svc(?::\d{1,5})?$`},
			Gateways:  []string{},
		},{
			URI:       &istiov1alpha1.StringMatch{Regex: "^/pets/(.*?)?"},
			Authority: &istiov1alpha1.StringMatch{Regex: `^test-route\.test-ns(\.svc(\.cluster\.local)?)?(?::\d{1,5})?$`},
			Gateways:  []string{},
		}},
```
Do we need to remove the short names from hosts in ingress rule or just add the additional match?
I'm not sure which is more reasonable.

Reply:
We should only keep the first rule, since it covers the next two cases.

Reply:
@tcnghia Current changes only keep "test-route.test-ns.svc.cluster.local".
Could this be merged or any other changes needed?